src/rules/commentFormatRule.ts

```diff
@@ -178,27 +201,78 @@ function walk(ctx: Lint.WalkContext<Options>) {
         }
 
         if (ctx.options.case === Case.None ||
             ctx.options.exceptions !== undefined && ctx.options.exceptions.test(commentText) ||
             ENABLE_DISABLE_REGEX.test(commentText)) {
+            addFailure(ctx, commentText, start, end, failures);
             return;
         }
 
         // search for first non-space character to check if lower or upper
         const charPos = commentText.search(/\S/);
         if (charPos === -1) {
+            addFailure(ctx, commentText, start, end, failures);
             return;
         }
-        if (ctx.options.case === Case.Lower) {
-            if (!isLowerCase(commentText[charPos])) {
-                ctx.addFailure(start, end, Rule.LOWERCASE_FAILURE + ctx.options.failureSuffix);
+        failures.firstLetterPos = charPos;
+        if (ctx.options.case === Case.Lower && !isLowerCase(commentText[charPos])) {
+            failures.lowercase = true;
+        } else if (ctx.options.case === Case.Upper && !isUpperCase(commentText[charPos])) {
+            failures.uppercase = true;
+            if (ctx.options.allowTrailingLowercase) {
+                const lineAndCharCurrentComment = ts.getLineAndCharacterOfPosition(ctx.sourceFile, pos);
+                const posPrevComment = ctx.sourceFile.getPositionOfLineAndCharacter(
+                    lineAndCharCurrentComment.line - 1, 0) + lineAndCharCurrentComment.character;
+
+                const prevComment = utils.getCommentAtPosition(ctx.sourceFile, posPrevComment);
+                if (prevComment !== undefined && prevComment.pos === posPrevComment) {
+                    failures.uppercase = false;
+                }
             }
-        } else if (!isUpperCase(commentText[charPos])) {
-            ctx.addFailure(start, end, Rule.UPPERCASE_FAILURE + ctx.options.failureSuffix);
         }
+        addFailure(ctx, commentText, start, end, failures);
     });
 }
+
+function addFailure(ctx: Lint.WalkContext<Options>, comment: string, start: number, end: number, failures: Failures) {
+    // No failure detected
+    if (!failures.leadingSpace && !failures.lowercase && !failures.uppercase) {
+        return;
+    }
+
+    if (failures.lowercase) {
+        const msg = failures.leadingSpace ? Rule.SPACE_LOWERCASE_FAILURE : Rule.LOWERCASE_FAILURE;
+        const fix = generateFix(comment, start, failures);
+        ctx.addFailure(start, end, msg + ctx.options.failureSuffix, fix);
+    } else if (failures.uppercase) {
+        const msg = failures.leadingSpace ? Rule.SPACE_UPPERCASE_FAILURE : Rule.UPPERCASE_FAILURE;
+        const fix = generateFix(comment, start, failures);
+        ctx.addFailure(start, end, msg + ctx.options.failureSuffix, fix);
+    } else {
+        // Only whitespace failure
+        ctx.addFailure(start, end, Rule.LEADING_SPACE_FAILURE, generateFix(comment, start, failures));
+    }
+}
+
+function generateFix(comment: string, start: number, failures: Failures) {
+    if (failures.lowercase) {
```

Comment:
Consider merging this with the function above. Currently you duplicate all conditions.